base/linked_list.jl

```diff
@@ -97,10 +97,10 @@ function list_deletefirst!(q::InvasiveLinkedList{T}, val::T) where T
             q.head = val.next::T
         end
     else
-        head_next = head.next
+        head_next = head.next::Union{T, Nothing}
```

Comment:
Why does inference have trouble figuring this one out?

Reply:
Hmm, that's a good question. I thought the type of the field was `Any`, but it's not. Nevertheless we get:
```
7 ── %23 = Base.getfield(%7, :next)::Union{Task, Nothing}
8 ┄─ %24 = φ (#7 => %7, #9 => %25)::Any
│    %25 = φ (#7 => %23, #9 => %29)::Any
│    %26 = (%25 === val)::Bool
│    %27 = Core.Intrinsics.not_int(%26)::Bool
└───       goto #10 if not %27
9 ── %29 = Base.getproperty(%25, :next)::Any
└───       goto #8
```